app/src/main/java/com/github/sdp/tarjetakuna/model/MagicSet.kt

```diff
@@ -1,30 +1,43 @@
-// ====================
-// MagicSet.kt
-// Tarjetakuna, 2023
-// ====================
-
 package com.github.sdp.tarjetakuna.model
 
+import com.github.sdp.tarjetakuna.utils.JsonAdapters
+import com.google.gson.annotations.JsonAdapter
+import java.time.LocalDate
+
 /**
  * Represents a Magic set.
  */
 data class MagicSet(
     /**
      * The set code.
      */
-    val code: String,
+    val code: String = "Unknown code",
 
     /**
      * The set name.
      */
-    val name: String
+    val name: String = "Unknown name",
+
+    /**
+     * The set type.
+     */
+    val type: String = "Unknown type",
+
+    /**
+     * The set block.
+     */
+    val block: String = "Unknown block",
+
+    /**
+     * The set release date.
+     */
+    @field:JsonAdapter(JsonAdapters.LocalDateAdapter::class)
+    val releaseDate: LocalDate = LocalDate.of(1970, 1, 1)
 ) {
     init {
```

Comment:
You initialize with default values that aren't blank, so isNotBlank will always pass. Blank values are (""," ", "\n","\t" etc). In the tests you manually input empty strings, although realistically I feel like we would take the approach of doing MagicSet(code = 'abc', name="xyz") and simply leaving out arguments that don't exist for that set (as that's the use of having default values). This also links to my question for the file SingleSetFragment.

Reply:
isNotBlank will always pass if we use the default arguments. I put arguments by default so as not to need to modify all the lines where we create a set when I added the type and block (and to have a healthier behavior just in case).

Reply:
Oh, so it will be possible to have a valid Set even though we might not have all the info about it? I think that is the part I didn't understand.

Reply:
this is a strange requirement indeed, as the default values are not blank. In what case is your require useful ?

Reply:
Preconditions are useful if we write (unintentionally) false data. For example, the number of health points of a character at -10. Here, default values will always and fortunately pass. However, if in our code, we make the mistake of not putting a name on the card, the require will detect it. It's the same for the block and the type.

Reply:
Well, that s the point, if you forget to put the name, it will not detect it. It's only if you put a blank name that this will be detected right?